test/integration/framework/perf_utils.go

```diff
@@ -125,7 +125,7 @@ func (p *IntegrationTestNodePreparer) CleanupNodes(ctx context.Context) error {
 	}
 	var errRet error
 	for i := range nodes.Items {
-		if err := p.client.CoreV1().Nodes().Delete(ctx, nodes.Items[i].Name, metav1.DeleteOptions{}); err != nil {
+		if err := p.client.CoreV1().Nodes().Delete(context.Background(), nodes.Items[i].Name, metav1.DeleteOptions{}); err != nil {
```

Comment:
Shouldn't callers to `CleanupNodes` pass the appropriate context?
Otherwise maybe it's worth removing `ctx` from the parameters.

Reply:
Didn't find an appropriate context, `b.Cleanup()` will be called when all its subtests complete, I tried to build a context at the top of `BenchmarkPerfScheduling`, but will cause other failures. 
```
F0105 23:56:05.103306   81606 etcd.go:217] unexpected number of goroutines: before: 2 after 6
```
Let's remove the ctx completely to solve the benchmark failure for now.

Reply:
Yeah, that's better
/lgtm
/approve

Reply:
Yes, that's what I want at the end, but as I pasted above, we'll have some errors then. So I'd like to merge this first to fix the CI error and then refactor as what we want together with some other works in `mustSetupScheduler` etc..
```
F0105 23:56:05.103306   81606 etcd.go:217] unexpected number of goroutines: before: 2 after 6
```

Make sense?

Reply:
The scheduler_perf has already had that `unexpected num of goroutines` error. 

```
unexpected number of goroutines: before: 2 after 43
```

https://storage.googleapis.com/kubernetes-jenkins/logs/ci-benchmark-scheduler-perf-master/1610954337838698496/build-log.txt

So, I'm not sure if passing the appropriate `ctx` really caused this error on your laptop.

Reply:
> So I'd like to merge this first to fix the CI error 

Sure, but why not do it "properly"? Removing the `ctx` parameter from `CleanupNodes` is going in the wrong direction. Just keep that as-is and change the cleanup code:
```
			b.Cleanup(func() {
                                // This cannot use the ctx above because it will be canceled
                                // by the time that this callback is invoked.
				_ = nodePreparer.CleanupNodes(context.Background())
```

Reply:
@sanposhiho Yes..., it seems like the error is another case. So I'll do more work.

The problem about goroutine leakage is tracked by https://github.com/kubernetes/kubernetes/issues/114890

Reply:
@pohly would you mind to take another look?

Reply:
> The scheduler_perf has already had that `unexpected num of goroutines` error.

The test seems to use "start" functions without a corresponding "wait" function, so the caller cannot be certain that all goroutines have really terminated before returning from the test.